src/ApiService/ApiService/OneFuzzTypes/Validated.cs

```diff
@@ -1,161 +1,107 @@
-    }
-
-    public static bool TryParse(string input, [NotNullWhen(returnValue: true)] out PoolName? result) {
-        if (!IsValid(input)) {
-            result = default;
-            return false;
-        }
-
-        result = new PoolName(input);
-        return true;
-    }
-
-    public sealed class Converter : ValidatedStringConverter<PoolName> {
-        protected override bool TryParse(string input, out PoolName? output)
-            => PoolName.TryParse(input, out output);
-    }
+    public static bool IsValid(string input) => true;
 }
 
-[JsonConverter(typeof(Converter))]
-public record Region : ValidatedString {
-    private static bool IsValid(string input) => Check.IsAlnum(input);
-
-    private Region(string value) : base(value.ToLowerInvariant()) {
-        Debug.Assert(IsValid(value));
-    }
-
-    public static Region Parse(string input) {
-        if (TryParse(input, out var result)) {
-            return result;
-        }
-
-        throw new ArgumentException("Region name must have only numbers or letters");
-    }
-
-    public static bool TryParse(string input, [NotNullWhen(returnValue: true)] out Region? result) {
-        if (!IsValid(input)) {
-            result = default;
-            return false;
-        }
-
-        result = new Region(input);
-        return true;
-    }
-
+[JsonConverter(typeof(ValidatedStringConverter<Region>))]
+public sealed record Region : ValidatedStringBase<Region>, IValidatedString<Region> {
+    private Region(string value) : base(value.ToLowerInvariant()) { }
+    public static Region Parse(string input) => new(input);
+    public static bool IsValid(string input) => Check.IsAlnum(input);
+    public static string Requirements => "Region name must have only numbers or letters";
     public static implicit operator AzureLocation(Region me) => new(me.String);
     public static implicit operator Region(AzureLocation it) => new(it.Name);
-
-    public sealed class Converter : ValidatedStringConverter<Region> {
-        protected override bool TryParse(string input, out Region? output)
-            => Region.TryParse(input, out output);
-    }
 }
 
-[JsonConverter(typeof(Converter))]
-public record Container : ValidatedString {
+[JsonConverter(typeof(ValidatedStringConverter<Container>))]
+public sealed record Container : ValidatedStringBase<Container>, IValidatedString<Container> {
+    private Container(string value) : base(value) { }
+    public static Container Parse(string input) => new(input);
+
     // See: https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftstorage
     // - 3-63
     // - Lowercase letters, numbers, and hyphens.
     // - Start with lowercase letter or number. Can't use consecutive hyphens.
     private static readonly Regex _containerRegex = new(@"\A(?!-)(?!.*--)[a-z0-9\-]{3,63}\z", RegexOptions.Compiled);
```

Comment:
Should we add this regex to `Check` too?

Reply:
Will do